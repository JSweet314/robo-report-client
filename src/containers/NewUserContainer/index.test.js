import React from 'react';
import { shallow } from 'enzyme';
import UserForm from './index';

describe('UserForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<UserForm />);
  });

  it('should match a snapshot with an instructions prompt', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match a snapshot with the form itself', () => {
    wrapper.setState({ isPrompted: false });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the change of form inputs', () => {
    expect(wrapper.state('email')).toEqual('');

    const mockEvent = { target: { id: 'email', value: 'test' }};
    
    wrapper.instance().handleOnChange(mockEvent);

    expect(wrapper.state('email')).toEqual('test');
  });

  it('should toggle whether the user is prompted with instructions', () => {
    expect(wrapper.state('welcomeDisplayed')).toEqual(true);

    wrapper.instance().toggleWelcome();

    expect(wrapper.state('welcomeDisplayed')).toEqual(false);
  });
});