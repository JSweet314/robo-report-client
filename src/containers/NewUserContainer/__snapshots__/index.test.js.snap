// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should match a snapshot with an instructions prompt 1`] = `
<NewUserWelcome
  toggleWelcome={[Function]}
/>
`;

exports[`UserForm should match a snapshot with the form itself 1`] = `
<NewUserWelcome
  toggleWelcome={[Function]}
/>
`;
